php/libraries/Data/DAO.php

```diff
@@ -0,0 +1,22 @@
+<?php
+namespace LORIS\Data;
+
+interface DAO
+{
+    /**
+     * Create a new DTO instance
+     *
+     * @return VO
+     */
+    public function getObject(array $primary_key);
+
+    /**
+     * Update the database with new values
+     *
+     * @param DTO $modifiedObject the modified object.
+     *
+     * @return void
+     */
+    public function update(DTO $modifiedObject);
```

Comment:
Why does a data access object have an update method? Those seem like they should be different concerns.

(I'd also consider changing it to something like "save" and clarifying what happens when a record doesn't exist..)